### PR TITLE
Backport v1.6.1 fixes

### LINKED ISSUE #3113: Invalid results for `find` with `filter` and `limit`
### FerretDB version

#2921 / #2993 introduced a regression in v1.6.0 for the `find` command with set `filter` and `limit` parameters.

### Backend

PostgreSQL, any version

### Environment

Any

### What did you do?

Run `db.col.find(<filter>).limit(1)` or `db.col.findOne(<filter>)`. The filter should be non-empty.

### What did you expect to see?

The expected document or null is returned.

### What did you see instead?

The result is not determined: the expected document, some other document, or null could be returned.

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #3113.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_find.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 	// Sorting requires fetching all documents and sorting them in memory unless `EnableSortPushdown` is set.
 	// Limit pushdown is not applied when `sort` is set but `EnableSortPushdown` is not set.
 	// Skip pushdown is not supported yet, limit pushdown is not applied when `skip` is non-zero value.
-	if (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
+	if params.Filter.Len() == 0 && (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
```
Comment (chilagrow) [inline internal/handlers/pg/msg_find.go:79]: Oh my dear, that's a really serious mistake of mine :disappointed:

Comment (AlekSi) [reply internal/handlers/pg/msg_find.go:79]: @chilagrow Please take over this PR tomorrow. At the very least, we should update those comments and add tests. Maybe something else.

Comment (chilagrow) [reply internal/handlers/pg/msg_find.go:79]: Yes sure, tests, documentation and comments should be updated.

Do we plan a patch release tomorrow?

Comment (AlekSi) [reply internal/handlers/pg/msg_find.go:79]: We will release v1.6.1 today, but that PR should be cleaned-up for v1.7.0

**Comment by chilagrow** [comment]
Testing combination of `-enable-sort-pushdown` and `-disable-filter-pushdown` flags

```
$ go test -count=1 -run=TestQueryCommandLimitPushDown . -enable-sort-pushdown=true -disable-filter-pushdown=true -target-backend=ferretdb-pg -target-tls -postgresql-url=postgres://username@127.0.0.1:5432/ferretdb?pool_max_conns=50 -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem'
ok      github.com/FerretDB/FerretDB/integration        0.280s
$ go test -count=1 -run=TestQueryCommandLimitPushDown . -enable-sort-pushdown=true -disable-filter-pushdown=false -target-backend=ferretdb-pg -target-tls -postgresql-url=postgres://username@127.0.0.1:5432/ferretdb?pool_max_conns=50 -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem'
ok      github.com/FerretDB/FerretDB/integration        0.289s
$ go test -count=1 -run=TestQueryCommandLimitPushDown . -enable-sort-pushdown=false -disable-filter-pushdown=false -target-backend=ferretdb-pg -target-tls -postgresql-url=postgres://username@127.0.0.1:5432/ferretdb?pool_max_conns=50 -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem'
ok      github.com/FerretDB/FerretDB/integration        0.302s
$ go test -count=1 -run=TestQueryCommandLimitPushDown . -enable-sort-pushdown=false -disable-filter-pushdown=true -target-backend=ferretdb-pg -target-tls -postgresql-url=postgres://username@127.0.0.1:5432/ferretdb?pool_max_conns=50 -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem'
ok      github.com/FerretDB/FerretDB/integration        0.302s
```

**Review by AlekSi — COMMENTED** [review]
LGTM

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by chilagrow — APPROVED** [review]

**Pull request merged**